apps/web/pages/_document.tsx

```diff
@@ -1,13 +1,25 @@
 import type { DocumentContext, DocumentProps } from "next/document";
 import Document, { Head, Html, Main, NextScript } from "next/document";
 import Script from "next/script";
+import { cloneElement } from "react";
 import { z } from "zod";
 
 import { getDirFromLang } from "@calcom/lib/i18n";
 
 import { csp } from "@lib/csp";
 
 type Props = Record<string, unknown> & DocumentProps;
+// So that bots without referrer header have the referrerpolicy set explicitly by the script tag
+class CustomHead extends Head {
+  getScripts(files: Parameters<Head["getScripts"]>[0]) {
+    const originalScripts = super.getScripts(files);
+    return originalScripts.map((script) => {
+      return cloneElement(script, {
+        referrerpolicy: "strict-origin-when-cross-origin",
```

Comment:
You can update this on the `next.config.js` as well. You don't need to create a new head components for this. 

An example next.config.mjs

```js
export default {
 headers() {
  return [{ source: '/(.*)', headers: [  {
    key: 'Referrer-Policy',
    value: 'origin-when-cross-origin',
  }]
 }
}

Reply:
Agreed!

Reply:
Oh, that's great actually. Much better, thank you @anonrig.

I also saw that it's possible to set the `referrerpolicy` on a document basis by a single `meta` element. Not if next rewriting the headers or adding this tag to the document make any difference. I will start with the header rewrite just as you proposed and if that doesn't work, I can try out the `meta` element.


```diff
// _document.tsx
 render() {
     //...
     return (
       <Html lang={locale} dir={dir}>
         <Head nonce={nonce}>
           {/* So that our rewrite logic depending on the `referrer` header is always set: */}
+         <meta name="referrer" content="strict-origin-when-cross-origin" />
           <link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png" />
```

Reply:
changed the approach to adding the `Referer` on the website repo instead: [see comment](https://github.com/calcom/cal.com/pull/8335#issuecomment-1513277529).